### ISSUE TITLE
Multiple Input Layers

### ISSUE THREAD

**Issue by lleuven** [viscom-ulm/Net2Vis#14]
Dear developers,

I really like your tool. For me, it is the very good way to visualize my networks. Currently, I am using a model with multiple input branches that are concatenated at a later stage. Unfortunately, Net2Vis assumes all input layers to be a split of the same input (even if the shapes differ). I would appreciate if your tool could support multiple input branches.

Please find attached the graphical output and my model code.

![grafik](https://user-images.githubusercontent.com/15818648/121022355-8b433900-c7a2-11eb-9669-e72d0b13515c.png)

```python
# You can freely modify this file.
# However, you need to have a function that is named get_model and returns a Keras Model.
import keras
from keras import models
from keras import layers
from keras import utils

def get_model():
    window = 65
    variables = 9

    n_branches = 4
    
    input_shape = [(window, 1, variables),
        (window, 1, variables),
        (window, 1, variables),
        (window, 1, variables + 1)]
    output_shape = 4

    conf = [64, 32, 16]
    bn = False
    activation = keras.layers.ReLU
    activation_output = keras.layers.Activation
    dropout = True
    dropout_rate = 0.2

    # input tail 0
    x_input_b0 = keras.layers.Input(shape=input_shape[0])
    x_in_b0 = keras.layers.Flatten()(x_input_b0)
    for layer, n_hidden in enumerate(conf):
        x_in_b0 = keras.layers.Dense(n_hidden)(x_in_b0)
        if bn is True:
            x_in_b0 = keras.layers.BatchNormalization()(x_in_b0)
        x_in_b0 = activation()(x_in_b0)
        if dropout is not None:
            x_in_b0 = keras.layers.Dropout(dropout_rate)(x_in_b0)

    # input tail 1
    x_input_b1 = keras.layers.Input(shape=input_shape[1])
    x_in_b1 = keras.layers.Flatten()(x_input_b1)
    for layer, n_hidden in enumerate(conf):
        x_in_b1 = keras.layers.Dense(n_hidden)(x_in_b1)
        if bn is True:
            x_in_b1 = keras.layers.BatchNormalization()(x_in_b1)
        x_in_b1 = activation()(x_in_b1)
        if dropout is not None:
            x_in_b1 = keras.layers.Dropout(dropout_rate)(x_in_b1)

    # input tail 2
    x_input_b2 = keras.layers.Input(shape=input_shape[2])
    x_in_b2 = keras.layers.Flatten()(x_input_b2)
    for layer, n_hidden in enumerate(conf):
        x_in_b2 = keras.layers.Dense(n_hidden)(x_in_b2)
        if bn is True:
            x_in_b2 = keras.layers.BatchNormalization()(x_in_b2)
        x_in_b2 = activation()(x_in_b2)
        if dropout is not None:
            x_in_b2 = keras.layers.Dropout(dropout_rate)(x_in_b2)

    # input tail 3
    x_input_b3 = keras.layers.Input(shape=input_shape[3])
    x_in_b3 = keras.layers.Flatten()(x_input_b3)
    for layer, n_hidden in enumerate(conf):
        x_in_b3 = keras.layers.Dense(n_hidden)(x_in_b3)
        if bn is True:
            x_in_b3 = keras.layers.BatchNormalization()(x_in_b3)
        x_in_b3 = activation()(x_in_b3)
        if dropout is not None:
            x_in_b3 = keras.layers.Dropout(dropout_rate)(x_in_b3)

    # concat all inputs
    x_input = [x_input_b0, x_input_b1, x_input_b2, x_input_b3]
    x_concat = keras.layers.Concatenate()([x_in_b0, x_in_b1, x_in_b2, x_in_b3])
    
    # output tail
    n_neurons_concat = int(conf[-1]) * n_branches
    layer_concat = 0
    for exp in reversed(range(2, n_branches + 1)):
        n_neurons = output_shape ** exp
        if n_neurons < n_neurons_concat:
            layer_concat += 1
            x_concat = keras.layers.Dense(n_neurons)(x_concat)
            if bn is True:
                x_concat = keras.layers.BatchNormalization()(x_concat)
            x_concat = activation()(x_concat)
            if dropout is not None:
                x_concat = keras.layers.Dropout(dropout_rate)(x_concat)
    x_concat = keras.layers.Dense(output_shape)(x_concat)
    out = activation_output("linear")(x_concat)
    model = keras.Model(inputs=x_input, outputs=[out])
    
    return model
```

**Comment by Sparkier** [comment]
Thank you for submitting this Issue and sorry I could not get to it earlier. This should be fixed with this commit: 47523d758336747de3b7efc040efbcaa90db6aa5

**Comment by Sparkier** [comment]
I copied your code here: https://viscom.net2vis.uni-ulm.de/multi_in_paths, and this is how it now looks for me: 
<img width="445" alt="image" src="https://user-images.githubusercontent.com/5690524/121804267-c8497880-cc45-11eb-9b93-09ec38a5a121.png">
